torchvision/datasets/phototour.py

```diff
@@ -0,0 +1,178 @@
+
+    def download(self):
+        print_('\n-- Loading PhotoTour dataset: {}\n'.format(self.name))
+
+        if self._check_exists():
+            print_('# Found cached data {}'.format(self.data_file))
+            return
+
+        if not self._check_downloaded():
+            # download files
+            url = self.urls[self.name]
+            filename = url.rpartition('/')[2]
+            file_path = os.path.join(self.root, filename)
+
+            try:
+                os.makedirs(self.root)
+            except Exception:
+                    pass
+
+            print_('# Downloading {}\n# Downloading {}\n\nIt might take while.'
+                   ' Please grab yourself a coffee and relax.\n'
+                   .format(url, file_path))
+
+            os.system('wget {} -P {}'.format(url, self.root))
+            assert os.path.exists(file_path)
+
+            print_('# Extracting data {}\n'.format(self.data_down))
+
+            import zipfile
+            with zipfile.ZipFile(file_path, 'r') as z:
+                z.extractall(self.data_dir)
+            os.unlink(file_path)
+        else:
+            print_('# Dataset already extracted {}'.format(self.data_dir))
+
+        # process and save as torch files
+        print_('# Caching data {}'.format(self.data_file))
+
+        data_set = (
+            read_image_file(self.data_dir, self.image_ext, self.size, self.lens[self.name]),
+            read_info_file(self.data_dir, self.info_file),
+            read_matches_files(self.data_dir, self.matches_files)
+        )
+
+        with open(self.data_file, 'wb') as f:
+            torch.save(data_set, f)
+
+
+def print_(text):
+    print('\033[;1m{}\033[0;0m'.format(text))
+
+
+def read_image_file(data_dir, image_ext, img_sz, n):
+    """Return a Tensor containing the patches
+    """
+    def PIL2array(_img, img_size):
+        """Convert PIL image type to numpy 2D array
+        """
+        return np.array(_img.getdata(), dtype=np.uint8) \
+            .reshape(img_size, img_size)
+
+    def read_filenames(_data_dir, _image_ext):
+        """Return a list with the file names of the images containing the patches
+        """
+        files = []
+        # find those files with the specified extension
+        for file_dir in os.listdir(_data_dir):
+            if file_dir.endswith(_image_ext):
+                files.append(os.path.join(_data_dir, file_dir))
+        return sorted(files)  # sort files in ascend order to keep relations
+
+    images = []
+    list_files = read_filenames(data_dir, image_ext)
+
+    # use opencv to read  dataset
+    for file_path in list_files:
+        # load the image containing the patches, crop in 64x64 patches and
+        # reshape to the desired size (default: 64)
+        img = cv2.imread(file_path, cv2.IMREAD_GRAYSCALE)
+        for y in range(0, 1024, 64):
+            for x in range(0, 1024, 64):
+                patch = img[y: y + 64, x: x + 64]
+                if img_sz != 64:
```

Comment:
Wouldn't it be better to extract patches of a given image size, rather than 64x64 + resize? I think if someone wants to resize the images, they can do it themselves, and this gives them additional control over the algorithm.